src/main.rs

```diff
@@ -1,65 +1,85 @@
-        .insert_resource(Settings::default())
-        .insert_resource(CameraController::default())
-        .insert_resource(lib)
-        .insert_resource(map)
-        .add_startup_system(setup_system)
-        .add_system(settings_system.label("settings"))
-        .add_system(ui_system.label("ui"))
-        .add_system(camera_controller_system.label("camera_control"))
-        .add_system(camera_system.label("camera").after("camera_control"))
-        .add_system(map_chunk_view_system.label("chunk_view").after("camera"))
-        .add_system(
-            visibility_system
-                .label("visibility")
-                .after("chunk_view")
-                .after("settings"),
-        )
-        .add_system(animation_system.label("animation").after("visibility"))
-        .run();
+            let map = Map::load(File::open(map_path)?)?;
+            let lib = ElementLibrary::load(File::open(lib_path)?)?;
 
-    Ok(())
+            App::new()
+                .add_plugins_with(DefaultPlugins, |group| {
+                    group.add_before::<bevy::asset::AssetPlugin, _>(JarAssetIo::plugin(gfx_path))
+                })
+                .add_plugin(EguiPlugin)
+                .add_plugin(LogDiagnosticsPlugin::default())
+                .add_plugin(FrameTimeDiagnosticsPlugin::default())
+                .init_asset_loader::<TgamLoader>()
+                .insert_resource(Settings::default())
+                .insert_resource(CameraController::default())
+                .insert_resource(lib)
+                .insert_resource(map)
+                .add_startup_system(setup_system)
+                .add_system(settings_system.label("settings"))
+                .add_system(ui_system.label("ui"))
+                .add_system(camera_controller_system.label("camera_control"))
+                .add_system(camera_system.label("camera").after("camera_control"))
+                .add_system(map_chunk_view_system.label("chunk_view").after("camera"))
+                .add_system(
+                    visibility_system
+                        .label("visibility")
+                        .after("chunk_view")
+                        .after("settings"),
+                )
+                .add_system(animation_system.label("animation").after("visibility"))
+                .run();
+
+            Ok(())
+        }
+        Err(_e) => match _e {
+            Error::MissingOption(_) | Error::OptionWithoutAValue(_) => {
+                match get_maps_id(maps_path.join("gfx")) {
+                    Ok(maps_id) => Err(anyhow!(
+                        "Map isn't specified, following maps id are available :\n{:?}",
+                        maps_id
+                    )),
+                    Err(_e) => Err(anyhow!(
+                        "Map isn't specified, but no maps found in game_path specified."
+                    )),
+                }
+            }
+            _ => Err(anyhow!(_e)),
+        },
```

Comment:
there's no need for a nested `match`, you can unnest it:
```suggestion
        Err(Error::MissingOption(_) | Error::OptionWithoutAValue("map")) => {
            match get_map_ids(maps_path.join("gfx")) {
                Ok(map_ids) => Err(anyhow!(
                    "Map isn't specified, following maps id are available: \n{:?}",
                    map_ids
                )),
                Err(_e) => Err(anyhow!(
                    "Map isn't specified, but no maps found under provided game_path"
                )),
            }
        }
        Err(other) => Err(anyhow!(other)),
```

Reply:
Thanks, I didn't see how to write it at first (I'm a beginner in Rust and I think it's noticeable)